node/chainSimulator/chainSimulator.go

```diff
@@ -412,30 +421,118 @@ func (s *simulator) SetStateMultiple(stateSlice []*dtos.AddressState) error {
 	return nil
 }
 
-// SendTxAndGenerateBlockTilTxIsExecuted will the provided transaction and generate block
-func (s *simulator) SendTxAndGenerateBlockTilTxIsExecuted(txToSend *transaction.Transaction, maxNumOfBlockToGenerateWhenExecutingTx int) (*transaction.ApiTransactionResult, error) {
-	txHashHex, err := s.sendTx(txToSend)
+// SendTxAndGenerateBlockTilTxIsExecuted will send the provided transaction and generate block until the transaction is executed
+func (s *simulator) SendTxAndGenerateBlockTilTxIsExecuted(txToSend *transaction.Transaction, maxNumOfBlocksToGenerateWhenExecutingTx int) (*transaction.ApiTransactionResult, error) {
+	result, err := s.SendTxsAndGenerateBlocksTilAreExecuted([]*transaction.Transaction{txToSend}, maxNumOfBlocksToGenerateWhenExecutingTx)
 	if err != nil {
 		return nil, err
 	}
 
+	return result[0], nil
+}
+
+// SendTxsAndGenerateBlocksTilAreExecuted will send the provided transactions and generate block until all transactions are executed
+func (s *simulator) SendTxsAndGenerateBlocksTilAreExecuted(txsToSend []*transaction.Transaction, maxNumOfBlocksToGenerateWhenExecutingTx int) ([]*transaction.ApiTransactionResult, error) {
+	if len(txsToSend) == 0 {
+		return nil, errEmptySliceOfTxs
+	}
+	if maxNumOfBlocksToGenerateWhenExecutingTx == 0 {
+		return nil, errInvalidMaxNumOfBlocks
+	}
+
+	transactionStatus := make([]*transactionWithResult, 0, len(txsToSend))
+	for idx, tx := range txsToSend {
+		if tx == nil {
+			return nil, fmt.Errorf("%w on position %d", errNilTransaction, idx)
+		}
+
+		txHashHex, err := s.sendTx(tx)
+		if err != nil {
+			return nil, err
+		}
+
+		transactionStatus = append(transactionStatus, &transactionWithResult{
+			hexHash: txHashHex,
+			tx:      tx,
+		})
+	}
+
 	time.Sleep(delaySendTxs)
 
-	destinationShardID := s.GetNodeHandler(0).GetShardCoordinator().ComputeId(txToSend.RcvAddr)
-	for count := 0; count < maxNumOfBlockToGenerateWhenExecutingTx; count++ {
-		err = s.GenerateBlocks(1)
+	for count := 0; count < maxNumOfBlocksToGenerateWhenExecutingTx; count++ {
+		err := s.GenerateBlocks(1)
 		if err != nil {
 			return nil, err
 		}
 
-		tx, errGet := s.GetNodeHandler(destinationShardID).GetFacadeHandler().GetTransaction(txHashHex, true)
-		if errGet == nil && tx.Status != transaction.TxStatusPending {
-			log.Info("############## transaction was executed ##############", "txHash", txHashHex)
-			return tx, nil
+		txsAreExecuted := s.computeTransactionsStatus(transactionStatus)
+		if txsAreExecuted {
+			return getApiTransactionsFromResult(transactionStatus), nil
+		}
+	}
+
+	return nil, errors.New("something went wrong. Transaction(s) is/are still in pending")
+}
+
+func (s *simulator) computeTransactionsStatus(status []*transactionWithResult) bool {
```

Comment:
can you rename `status` in `txsWithResult` or something similar

Reply:
renamed